### PR TITLE
feat(web): add org credit and invoice page procedures

### PR DESCRIPTION
**Pull request by iscekic**
## Summary

No new behavior. The two new paginated endpoints are not consumed by any page yet, so a user sees the same organization credit transaction and invoice views as before.

---

## Reviewer Notes

The organization credit transaction query gains a cursor-paginated form, the `CreditTransactionsPage` contract. `getCreditTransactionsForOrganizationPage` returns 25 entries plus a probe ordered newest first, with a `nextCursor`, a `hasMore` flag, and a `CreditSummary` computed by the new `getCreditTransactionsSummaryForOrganization`; both helpers exclude `kpo:consumption:` categories exactly like the old query. The old `getCreditTransactionsForOrganization` array form stays unchanged and is marked `old form`, so existing callers keep working until every client migrates to the page form.

<details>
<summary>Files</summary>

- `apps/web/src/lib/creditTransactions.ts` — exports the `CreditSummary` type; adds `getCreditTransactionsSummaryForOrganization` and `getCreditTransactionsForOrganizationPage`; marks the old `getCreditTransactionsForOrganization` with an `old form` comment.

</details>

The Stripe invoice listing gains a cursor-paginated form, the `StripeInvoicesPage` contract. `getStripeInvoicesPage` lists 25 invoices with an optional `starting_after` cursor, reads `has_more`, and returns a `nextCursor` equal to the last invoice id. The invoice-to-unified mapping is extracted into `mapStripeInvoicesToUnified` and reused by both functions; the old `getStripeInvoices` keeps returning a flat array of up to 100 and is marked `old form`.

<details>
<summary>Files</summary>

- `apps/web/src/lib/stripe/index.ts` — adds the `StripeInvoicesPage` type, the `mapStripeInvoicesToUnified` helper, and `getStripeInvoicesPage`; marks `getStripeInvoices` with an `old form` comment.

</details>

Two new tRPC procedures expose the paginated queries. `organizations.creditTransactionsPage` serves the member role with a defaulted non-negative numeric cursor, and `organizations.invoicesPage` serves the billing role with an optional string cursor, resolving or creating the Stripe customer id first just like the old `invoices` procedure. The existing `creditTransactions` and `invoices` procedures keep their array returns, so no client breaks.

<details>
<summary>Files</summary>

- `apps/web/src/routers/organizations/organization-router.ts` — adds the `OrganizationTransactionsPageInputSchema` and `OrganizationInvoicesPageInputSchema` schemas plus the `creditTransactionsPage` and `invoicesPage` procedures beside their array-returning predecessors.

</details>

Tests: 2 test files changed — `creditTransactions.page.test.ts` (new) and `stripe/index.test.ts` (extended); both DB-backed tests need the test Postgres and run in CI.
Generated: none.

---

## Verification

- [ ] Manual tests not run on this host: the DB-backed tests require the test Postgres (Docker) and run in CI.
- E2E report: none attached. E2E runs once on the tip PR, not per stack level.

Human steps: none. No new environment values, secrets, or migrations.

## Visual Changes

Visual Changes: N/A

---

**Stacked PRs — merge bottom to top.** Each level shows only its own diff.

Full verification (E2E, user advocacy, simplify, bot review) runs on the tip PR over every level.
A finding on a level is repaired on that level, then carried upward with stack.sh forward.

1. `audit-w7b-data-contracts-dffd` — #5460
2. `audit-w7b-data-contracts-dffd-s2` — #5462 ← this PR
3. `audit-w7b-data-contracts-dffd-s3` — #5468
4. `audit-w7b-data-contracts-dffd-s4` — #5472
5. `audit-w7b-data-contracts-dffd-s5` — #5479 (tip)

### CODE REVIEW

File: apps/web/src/lib/creditTransactions.ts [hunk 1]
```diff
@@ -100,6 +128,71 @@ export async function getCreditTransactionsForOrganization(organizationId: Organ
...
+          eq(credit_transactions.organization_id, organizationId),
+          or(
+            isNull(credit_transactions.credit_category),
+            notLike(credit_transactions.credit_category, 'kpo:consumption:%')
+          )
+        )
+      )
+      .orderBy(desc(credit_transactions.created_at), desc(credit_transactions.id))
+      .limit(CREDIT_TRANSACTIONS_PAGE_SIZE + 1)
+      .offset(cursor),
```
Comment (pandemicsyn) [inline apps/web/src/lib/creditTransactions.ts]: bot: The advertised cursor is an OFFSET, so ledger pages are not stable when credit transactions change between requests.

Suggested fix: Encode the last returned `(created_at, id)` as the cursor and query for rows strictly after that boundary in the descending order (with the same filters), rather than using `.offset(cursor)`. This keeps later pages disjoint when new transactions arrive or earlier rows are removed. Add a test that fetches page 1, changes the ordered result set, then verifies page 2 does not duplicate or omit the original continuation.

File: apps/web/src/lib/stripe/index.ts [hunk 2]
```diff
@@ -703,6 +708,44 @@ export async function getStripeInvoices(
...
+  }
+
+  const invoices = await client.invoices.list(listParams);
+  const entries = mapStripeInvoicesToUnified(invoices.data);
+  const lastInvoice = invoices.data[invoices.data.length - 1];
+
+  return {
+    entries,
+    hasMore: invoices.has_more,
+    nextCursor: lastInvoice ? lastInvoice.id : null,
```
Comment (pandemicsyn) [inline apps/web/src/lib/stripe/index.ts]: bot: A non-empty final Stripe page returns a continuation cursor even when Stripe says there are no more invoices.

Suggested fix: Tie the cursor to Stripe's continuation signal, e.g. `nextCursor: invoices.has_more ? lastInvoice?.id ?? null : null`, and add a test with non-empty `data` plus `has_more: false` that expects `nextCursor` to be null.

**Review by pandemicsyn — APPROVED** [review]
